src/lib/Libattr/attr_fn_entlim.c

```diff
@@ -912,14 +912,15 @@ set_entlim(attribute *old, attribute *new, enum batch_op op)
 			/* having removed one or more elements from the value tree */
 			/* see if any entries are left or if the value is now null */
 			pkey = NULL;
-			if (entlim_get_next(pkey, oldctx) == NULL) {
+			if ((pkey = entlim_get_next(pkey, oldctx)) == NULL) {
 				/* no entries left set, clear the entire attribute */
 				free_entlim(old);
 				/* set _MODIFY flag so up level functions */
 				/* know the attribute has been changed    */
 				old->at_flags |= ATR_VFLAG_MODIFY;
 				return (0);
 			}
+			free(pkey);
```

Comment:
Should we also free(pkey) in the two while loops occurring in this function?

Reply:
pkey memory allocation happens at the beginning of the tree traversal. It is freed when the traversal is complete. But if we return the control from the loop in between, the memory has to be freed. I've addressed this with my recent commit. Please check.